### PR TITLE
Fix some straightforward PhanUndeclared issues

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

Mostly it's adding suppressions for false positives, and a few phpdoc updates, `@phan-var` type assertions, and test updates.

There are some functional changes though:

* packages/forms: Add a missing `use Jetpack_Options`.
* packages/identity-crisis: Add missing `use` of some Connection classes.
* packages/jetpack-mu-wpcom: Remove some references to wpcom stuff that was removed from wpcom itself a year+ ago.
* packages/my-jetpack: Define `module_name` property on `Hybrid_Product` (defaulting to null). Then replace some `empty()` checks with boolean checks.
* packages/plans: Add missing `use Store_Product_List`.
* packages/search: Deprecate `Search_Widget->jetpack_search` prop that has never been set since the package was created.
* packages/sync: Fix `Replicastore::get_user()` to actually work.

I left Jetpack for a separate PR, it looks to have a long list of fixes being needed.

I also left CRM, WP Super Cache, and VaultPress alone for now thanks to all the phpcs exclusions that will probably make any fixes have to go down a rabbit hole of dealing with phpcs-changed.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

None

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* CI happy?
* Stuff still work?

### CODE REVIEW

File: projects/packages/config/.phan/baseline.php [hunk 1]
```diff
@@ -8,12 +8,9 @@
  * (can be combined with --load-baseline)
  */
 return [
-    // # Issue statistics:
-    // PhanUndeclaredClassMethod : 1 occurrence
-
+    // This baseline has no suppressions
```
Comment (tbradsha) [inline projects/packages/config/.phan/baseline.php:11]: 🥳

File: projects/packages/connection/tests/php/test-rest-endpoints.php [hunk 2]
```diff
@@ -1816,6 +1816,5 @@ private function shutdown_reconnect_test( $invalid_token ) {
 		}
 
 		remove_filter( 'jetpack_options', array( $this, 'mock_jetpack_options' ), 10 );
-		remove_filter( 'pre_http_request', array( $this, 'intercept_validate_tokens_request' ), 10 );
```
Comment (tbradsha) [inline projects/packages/connection/tests/php/test-rest-endpoints.php:1819]: Why remove this?

File: projects/packages/connection/tests/php/test_jetpack_xmlrpc_server.php [hunk 3]
```diff
@@ -271,7 +271,6 @@ public function test_remote_connect_fails_no_blog_token() {
 
 		$server = new Jetpack_XMLRPC_Server();
 
-		add_filter( 'pre_http_request', array( $this, '__return_token' ) );
```
Comment (tbradsha) [inline projects/packages/connection/tests/php/test_jetpack_xmlrpc_server.php:274]: Why remove this?

File: projects/packages/connection/tests/php/test-rest-endpoints.php (hunk 2 above)
Comment (tbradsha) [reply projects/packages/connection/tests/php/test-rest-endpoints.php:1819]: Never mind, I see that function doesn't exist (as per the changelog).

File: projects/packages/connection/tests/php/test_jetpack_xmlrpc_server.php (hunk 3 above)
Comment (tbradsha) [reply projects/packages/connection/tests/php/test_jetpack_xmlrpc_server.php:274]: Never mind, I see that function doesn't exist (as per the changelog).

File: projects/packages/search/src/widgets/class-search-widget.php [hunk 4]
```diff
@@ -38,10 +38,11 @@ class Search_Widget extends \WP_Widget {
 	 */
 	const DEFAULT_SORT = 'relevance_desc';
 	/**
-	 * The Jetpack_Search instance.
+	 * Never used.
```
Comment (tbradsha) [inline projects/packages/search/src/widgets/class-search-widget.php:41]: Is it not used here?
https://github.com/Automattic/jetpack/blob/9d06de7b232ee5a227ce8eaf5a5bf704d78340b8/projects/packages/search/src/initializers/class-initializer.php#L207

...via here?
https://github.com/Automattic/jetpack/blob/9d06de7b232ee5a227ce8eaf5a5bf704d78340b8/projects/packages/search/src/initializers/class-initializer.php#L112

File: projects/packages/sync/src/class-replicastore.php [hunk 5]
```diff
@@ -1086,10 +1086,11 @@ public function user_count() {
 	 * @access public
 	 *
 	 * @param int $user_id User ID.
-	 * @return \WP_User User object.
+	 * @return \WP_User|null User object, or `null` if user invalid/not found.
```
Comment (tbradsha) [inline projects/packages/sync/src/class-replicastore.php:1089]: Not a blocker, but probably should update the doc in the interface as well?

**Comment by tbradsha** [comment]
> I also left CRM [...] alone for now thanks to all the phpcs exclusions that will probably make any fixes have to go down a rabbit hole of dealing with phpcs-changed.

I can guarantee this is the case. :^)

I left one comment about updating the `Replicastore_Interface->get_user()` doc to match its class implementation. Though looking further, the interface doesn't have the `@return` anyway, so happy to approve if you think it's good enough.

File: projects/packages/sync/src/class-replicastore.php (hunk 5 above)
Comment (anomiex) [reply projects/packages/sync/src/class-replicastore.php:1089]: Good suggestion. 2d26deef3c20ea71a1e882761fe7833f44f2e4b0

**Review by tbradsha — APPROVED** [review]

**Pull request merged**